node/week2/app.js

```diff
@@ -0,0 +1,94 @@
+import express from 'express';
+import fs from 'fs/promises';
+
+const app = express();
+const port = process.env.PORT || 3000;
+
+app.use(express.json());
+
+async function getDocuments() {
+  try {
+    const data = await fs.readFile('/workspace/class29-homework/node/week2/document.json', 'utf-8');
+    return JSON.parse(data);
+  } catch (error) {
+    console.error("Error reading documents:", error);
+    throw new Error("Could not load documents.");
+  }
+}
+
+app.get("/", (req, res) => {
+  res.send("This is a search engine");
+});
+
+app.get("/search", async (req, res) => {
+  const { q } = req.query;
+  try {
+    const documents = await getDocuments();
+
+    if (!q) {
+      return res.json(documents);
+    }
+
+    const filteredDocuments = documents.filter((doc) =>
+      Object.values(doc).some(value =>
+        String(value).toLowerCase().includes(q.toLowerCase())
+      )
+    );
```

Comment:
Instead of using `Object.values(doc)` it could be a good idea to explicitly choose the fields which should be searched. Then you are also making sure that you don't have to cast to `String`

Reply:
Dear mentor, changes made as per suggestions.